ocaml/typing/typedtree.mli

```diff
@@ -209,6 +209,12 @@ and exp_extra =
         (** Used for method bodies. *)
   | Texp_newtype of string * Jkind.annotation option
         (** fun (type t : immediate) ->  *)
+  | Texp_mode_coerce of Jane_syntax.Mode_expr.t
+        (** local_ E *)
+
+(* CR modes: Consider fusing [Texp_mode_coerce] and [Texp_constraint] when
+   the syntax changes.
+*)
```

Comment:
I imagine mode coercion and type coercion in principle don't need to show up together, and therefore should be separated here.

(Sure, our current syntax only supports `exp : ty @@ modes` which means modes can only occur accompanied by types, but I think we shouldn't let concrete syntax affect the representation of typed tree.)

Reply:
Let's make this decision (and delete this CR) once we have the new syntax.